salt/states/panos.py

```diff
@@ -0,0 +1,578 @@
+
+:codeauthor: :email:`Spencer Ervin <spencer_ervin@hotmail.com>`
+:maturity:   new
+:depends:    none
+:platform:   unix
+
+
+About
+=====
+This state module was designed to handle connections to a Palo Alto based
+firewall. This module relies on the Palo Alto proxy module to interface with the devices.
+
+This state module is designed to give extreme flexibility in the control over XPATH values on the PANOS device. It
+exposes the core XML API commands and allows state modules to chain complex XPATH commands.
+
+Below is an example of how to construct a security rule and move to the top of the policy. This will take a config
+lock to prevent execution during the operation, then remove the lock. After the XPATH has been deployed, it will
+commit to the device.
+
+.. code-block:: yaml
+
+    panos/takelock:
+        panos.add_config_lock
+    panos/service_tcp_22:
+        panos.set_config:
+            - xpath: /config/devices/entry[@name='localhost.localdomain']/vsys/entry[@name='vsys1']/service
+            - value: <entry name='tcp-22'><protocol><tcp><port>22</port></tcp></protocol></entry>
+            - commit: False
+    panos/create_rule1:
+        panos.set_config:
+            - xpath: /config/devices/entry[@name='localhost.localdomain']/vsys/entry[@name='vsys1']/rulebase/security/rules
+            - value: '
+              <entry name="rule1">
+                <from><member>trust</member></from>
+                <to><member>untrust</member></to>
+                <source><member>10.0.0.1</member></source>
+                <destination><member>10.0.1.1</member></destination>
+                <service><member>tcp-22</member></service>
+                <application><member>any</member></application>
+                <action>allow</action>
+                <disabled>no</disabled>
+              </entry>'
+            - commit: False
+    panos/moveruletop:
+        panos.move_config:
+            - xpath: /config/devices/entry[@name='localhost.localdomain']/vsys/entry[@name='vsys1']/rulebase/security/rules/entry[@name='rule1']
+            - where: top
+            - commit: False
+    panos/removelock:
+        panos.remove_config_lock
+    panos/commit:
+        panos.commit
+
+.. seealso::
+    :prox:`Palo Alto Proxy Module <salt.proxy.panos>`
+
+"""
+# Import Python Libs
+from __future__ import absolute_import
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    return 'panos.commit' in __salt__
+
+
+def _default_ret(name):
+    """
+    Set the default response values.
+
+    """
+    ret = {
+        'name': name,
+        'changes': {},
+        'commit': None,
+        'result': False,
+        'comment': ''
+    }
+    return ret
+
+
+def add_config_lock(name):
+    """
+    Prevent other users from changing configuration until the lock is released.
+
+    name: The name of the module function to execute.
+
+    SLS Example:
+
+    .. code-block:: bash
```

Comment:
I am being very picky, but this should be `yaml` instead of `bash` for the example label as they are sls examples. There are a couple more places below where this should also be changed.